react-icons for Social links and footer and Discord Added

nijamaDev opened the pull request:
Made a simple change so the icons in the social links use react-icons instead of having the pure svg defined in the file, also changed the icons import in the footer to have cleaner and more consistent code.

List of changes:
- Replaced '@react-icons/all-files' with updated 'react-icons'
- Cleaner imports for icons
- Use react-icons components instead of pure svg for the social links
- Added **Discord** to the social links and footer
- Better comments in site.config.ts

GitMac32 commented:
this code is full of garbage bloat & security concerns. 

Restricting EsLint from proper checks
Intentionally setting a security policy restricting script sources
checking for runtime env in gitactions... 

I could go on here, but I don't have to.  

transitive -bullshit is right. shame on you all

nijamaDev commented:
Excuse me? What are you talking about?

I think you might be in the wrong PR. I just added a way of setting up discord to the social links and on the footer and updated the react icons

nijamaDev commented:
> this code is full of garbage bloat & security concerns.
> 
> Restricting EsLint from proper checks Intentionally setting a security policy restricting script sources checking for runtime env in gitactions...
> 
> I could go on here, but I don't have to.
> 
> transitive -bullshit is right. shame on you all

Excuse me? What are you talking about?

I think you might be in the wrong PR. I just added a way of setting up discord to the social links and on the footer and updated the react icons

transitive-bullshit commented:
Thanks @nijamaDev. This PR looks great. Sorry I don't have time to maintain this project properly and I really have no idea what the other user was talking about.

I'm going through and doing a bunch of updates at the moment, so will consider these changes in a bit.
